pkg/catalog/routes.go

```diff
@@ -556,12 +556,23 @@ func (mc *MeshCatalog) buildInboundPolicies(t *access.TrafficTarget) []*trafficp
 
 		servicePolicy := trafficpolicy.NewInboundTrafficPolicy(buildPolicyName(destService, false), hostnames)
 
-		weightedCluster := getDefaultWeightedClusterForService(destService)
-
-		for _, sourceServiceAccount := range trafficTargetIdentitiesToSvcAccounts(t.Spec.Sources) {
-			for _, routeMatch := range routeMatches {
-				servicePolicy.AddRule(*trafficpolicy.NewRouteWeightedCluster(routeMatch, weightedCluster), sourceServiceAccount)
+		// while building inbound policies, check if the service is an apex service in traffic split
+		if mc.isApexService(destService) {
+			var localDestServices []service.MeshService
+			// build the inbound policy for an apex service to have its weighted clusters point to the backend services
+			backendServices := mc.getBackendServiceForApexService(destService)
+			// among all the backedn services, select only those that operlap with the destination services
+			// (i.e the destination service account)
```

Comment:
`destination services (i.e. destination service account)` is confusing. Maybe rephrase.

Reply:
removed `(i.e. destination service account)` . Hopefully that reduces the confusion